github/repos_projects.go

```diff
@@ -0,0 +1,461 @@
+	u := fmt.Sprintf("repos/%v/%v/projects/%v", owner, repo, number)
+	req, err := s.client.NewRequest("DELETE", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeProjectsPreview)
+
+	return s.client.Do(req, nil)
+}
+
+// ProjectColumn represents a column of a GitHub Project.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/projects/
+type ProjectColumn struct {
+	ID         *int       `json:"id,omitempty"`
+	Name       *string    `json:"name,omitempty"`
+	ProjectURL *string    `json:"project_url,omitempty"`
+	CreatedAt  *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt  *Timestamp `json:"updated_at,omitempty"`
+}
+
+// ListProjectColumns lists the columns of a GitHub Project for a repo.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/projects/#list-columns
+func (s *RepositoriesService) ListProjectColumns(owner, repo string, number int, opt *ListOptions) ([]*ProjectColumn, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/projects/%v/columns", owner, repo, number)
+	u, err := addOptions(u, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeProjectsPreview)
+
+	columns := []*ProjectColumn{}
+	resp, err := s.client.Do(req, &columns)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return columns, resp, err
+}
+
+// GetProjectColumn gets a column of a GitHub Project for a repo.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/projects/#get-a-column
+func (s *RepositoriesService) GetProjectColumn(owner, repo string, columnID int) (*ProjectColumn, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/projects/columns/%v", owner, repo, columnID)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeProjectsPreview)
+
+	column := &ProjectColumn{}
+	resp, err := s.client.Do(req, column)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return column, resp, err
+}
+
+// ProjectColumnInfo specifies the parameters to the
+// RepositoriesService.CreateProjectColumn and RepositoriesService.UpdateProjectColumn methods.
+type ProjectColumnInfo struct {
+	// The name of the project column. (Required for creation and update.)
+	Name string `json:"name"`
+}
+
+// CreateProjectColumn creates a column for the specified (by number) project.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/projects/#create-a-column
+func (s *RepositoriesService) CreateProjectColumn(owner, repo string, number int, columnInfo *ProjectColumnInfo) (*ProjectColumn, *Response, error) {
```

Comment:
In this case, `ProjectColumnInfo` has only one field and it's required (not optional). Yet `columnInfo` is a pointer instead of a value, so it can be `nil`.

I wanted to check if this was consistent with other parts of this library, and whether we could make it a value instead.

It appears to be very consistent, all similar structs are always passed as a pointer. I'm not sure why that's the case, since it seems like friendlier API to not give an option to pass nil when it's meaningless/unsupported. Perhaps it was done for performance.

In any case, this is fine, just wanted to comment on it.